### PR TITLE
Cascade between position record parameters and properties.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/44559

### CODE REVIEW

File: src/EditorFeatures/Test2/FindReferences/FindReferencesTests.ParameterSymbol.vb [hunk 1]
```diff
@@ -727,5 +727,173 @@ end class
...
+{
+    public int y { get; } = [|y|];
+}
+
+class P
+{
+    static void Main()
+    {
+        var f = new Goo(0, [|y|]: 1);
+        Console.WriteLine(f.y);
```
Comment (RikkiGibson) [inline src/EditorFeatures/Test2/FindReferences/FindReferencesTests.ParameterSymbol.vb:805]: A primary constructor parameter is deeply related to the member with the same name even when it is explicitly declared.  For example:

```cs
var rec = new Rec(1, 2);
var (x, y) = rec;


public record Rec(int x, int y)
{
    int x, y;
}
```

The synthesized Deconstruct is specified to read from the member that has the same name--that's the explicitly declared x and y fields in this case.

Given that, would it make sense to include the usage of 'y' here in the result set?

**Comment by RikkiGibson** [comment]
Talking offline, we think it would be ideal for the compiler to expose public APIs which let the IDE do the right thing here for both source and metadata scenarios.

For example, if a record `Rec` is declared in a referenced assembly:

```cs
var rec = new Rec(param: 1); // find-all-refs on "param" here
_ = rec.param.M(); // should include 'param' in results
```

I'm going to take a stab at coming up with public API that would address the scenario and we can bring it to API review.

File: src/EditorFeatures/Test2/FindReferences/FindReferencesTests.ParameterSymbol.vb (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test2/FindReferences/FindReferencesTests.ParameterSymbol.vb:805]: i'll wait on https://github.com/dotnet/roslyn/pull/59293#issuecomment-1030338868

if we get that, we can change this.

**Review by davidwengier — APPROVED** [review]
I think we'll still probably get complaints about this until the new API is available, but good to make at least some scenarios better in the meantime.

Does this also fix https://github.com/dotnet/roslyn/issues/50450? or even https://github.com/dotnet/roslyn/issues/44560 maybe?

**Comment by CyrusNajmabadi** [comment]
> Does this also fix https://github.com/dotnet/roslyn/issues/50450? or even https://github.com/dotnet/roslyn/issues/44560 maybe?

Very likely. I'll check today!

**Pull request merged**